chia/consensus/blockchain.py

```diff
@@ -259,6 +259,7 @@ async def receive_block(
             npc_result,
             fork_point_with_peak,
             self.get_block_generator,
+            pre_validation_result is None or not pre_validation_result.validated_signature,
```

Comment:
it's not obvious what this bool means, just reading this code. I have to look up the `validate_block_body()` function to ensure it's correct. It might be a good idea to name this parameter.

Reply:
there's a valid return valid pre_validation_result that says it didn't validate signatures? (it seems a bit risky and unintuitive)

Reply:
It's meant to convey that signatures were not validated so we must validate them later. Otherwise we might forget to validate them in the future. Block body validation does the safe thing and validates signatures if the result is False.

Reply:
right. I realized later you added this field. I suppose it makes sense to be explicit about this, rather than deduce it at this point by looking at the error code (and possibly something else).

This line in particular could be made clearer though, I think. You could give this value a name, like:

```
validate_signatures = pre_validation_result is None or not pre_validation_result.validated_signature
```

Reply:
> it's not obvious what this bool means, just reading this code. I have to look up the `validate_block_body()` function to ensure it's correct. It might be a good idea to name this parameter.

OK i will clean that up

Reply:
That is only None in tests, that is annoying.. I think it's worth cleaning up though

Reply:
In general, I would like to see named parameters roughly everywhere.  At one point I only half jokingly thought about setting my personal style guide to include `*` in every function def to force every parameter to be keyword only.